File: backend/src/AsliApp.Domain/Education/EducationModule.cs
namespace AsliApp.Domain.Education;

public sealed class EducationModule
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool IsPublished { get; set; }
    public DateTimeOffset CreatedAtUtc { get; set; }
    public DateTimeOffset UpdatedAtUtc { get; set; }

    public ICollection<Lesson> Lessons { get; set; } = [];
}

File: mobile/lib/features/education/domain/models/education_module.dart
import 'package:asli_app/features/education/domain/models/lesson.dart';

final class EducationModule {
  const EducationModule({
    required this.id,
    required this.title,
    required this.description,
    required this.order,
    required this.lessonCount,
    required this.lessons,
  });

  final String id;
  final String title;
  final String description;
  final int order;
  final int lessonCount;
  final List<Lesson> lessons;
}
